### ISSUE TITLE
Loading Ads using Async or Defer

### ISSUE THREAD

**Issue by ashrafulhaque89** [atwwei/atwwei.github.io#5]
Can the ads be loaded using async or defer? How to load ads without blocking the main thread or load ad properly in ngAfterViewInit?
![image](https://user-images.githubusercontent.com/38048752/179980546-6a791e8c-54be-4d1d-9a88-6d6301a8ac60.png)

**Comment by amcgranemyhome** [comment]
I would also like to have an answer to this - ads are really hurting performance, halving the https://pagespeed.web.dev/ score

**Comment by atwwei** [comment]
@ashrafulhaque89 @amcgranemyhome Sorry, I just saw this issue！

It's complicated to deal with. My suggestions：

1. [googletag.pubads().enableSingleRequest();](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/47de3e0f285ad7de9398c660b49a984fc5f78279/types/googletag/index.d.ts#L543-L554)
1. [googletag.pubads().enableLazyLoad();](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/47de3e0f285ad7de9398c660b49a984fc5f78279/types/googletag/index.d.ts#L501-L542)
1. [googletag.pubads().disableInitialLoad();](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/47de3e0f285ad7de9398c660b49a984fc5f78279/types/googletag/index.d.ts#L469-L478)
1. [GPT_LOADER](https://github.com/atwwei/atwwei.github.io/blob/5ad00d1259d3cefd0be1fbb3b9bbfbb935bba317/projects/dfp/src/app/app.module.ts#L27-L35)

**Comment by ashrafulhaque89** [comment]
@atwwei Thanks for your response. We have tried first 2 options. Will try the rest of the 2 options and update the issue.

**Comment by atwwei** [comment]
@ashrafulhaque89 I just added the `GPT_LOADER` sample code

https://github.com/atwwei/atwwei.github.io/blob/3f136ac6e9402530df0f00b7a66c54ffee0793e9/projects/dfp/src/app/app.module.ts#L26-L37

**Comment by ashrafulhaque89** [comment]
Will it help deferring the ad loading by adding above sample code (GPT_LOADER) to app.module.ts? @atwwei

**Comment by atwwei** [comment]
@ashrafulhaque89  Of course, you can also use any observable obejct to control the loading time by yourself.

**Comment by ashrafulhaque89** [comment]
Thanks 👍 @atwwei 
I will give it a try!
